OloEngine/tests/ShaderGraph/ShaderGraphSerializationTest.cpp

```
@@ -0,0 +1,199 @@
+
+TEST_F(ShaderGraphSerializationTest, RoundTripPreservesLinkCount)
+{
+    auto original = CreateSampleGraph();
+    std::string yaml = serializer.TestSerializeToYAML(original);
+
+    auto deserialized = Ref<ShaderGraphAsset>::Create();
+    ASSERT_TRUE(serializer.TestDeserializeFromYAML(yaml, deserialized));
+    EXPECT_EQ(deserialized->GetGraph().GetLinks().size(), original->GetGraph().GetLinks().size());
+}
+
+TEST_F(ShaderGraphSerializationTest, RoundTripPreservesParameterName)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)